lib/Compiler.js

```diff
@@ -110,7 +110,17 @@ Watching.prototype.watch = function(files, dirs, missing) {
 	}.bind(this));
 };
 
-Watching.prototype.invalidate = function() {
+Watching.prototype.invalidate = function(callback) {
+	callback = callback || function() {};
+	this.callbacks = this.callbacks || [];
+	this.callbacks.push(callback);
+	this._done = function() {
```

Comment:
don't *override*  the `_done` function. Just add this into the `_done` function.